src/cpp/StatisticsBackendData.cpp

```diff
@@ -69,21 +84,19 @@ StatisticsBackendData::~StatisticsBackendData()
 
 StatisticsBackendData* StatisticsBackendData::get_instance()
```

Comment:
I do not get what we would achieve. Passing a raw ptr from a smart ptr is discouraged, but it is also passing a unique_ptr.
If your idea is to use an `static instance` variable, there is a `reset` function that must change and it would become much complex.

Reply:
We will avoid someone accidentally doing something like:

```
StatisticsBackendData* data = StatisticsBackendData::get_instance();

...

delete data;
```

We will be returning a `const unique_ptr<StatisticsBackendData>&` which is API compatible with a raw pointer. This means that code using `auto data = StatisticsBackendData::get_instance()` will not need to be modified

Reply:
It changes because `auto` must change to `auto&`. But okey, I make the change.